executor/src/constant_evaluator/mod.rs

```diff
@@ -432,76 +453,68 @@ mod test {
     #[test]
     pub fn comparisons() {
         let src = r#"
-            constant %N = 6;
-            namespace std::convert(%N);
+            let N: int = 6;
+            namespace std::convert(N);
             let int = 9;
             let fe = 8;
-            namespace F(%N);
-            col fixed id(i) { i };
-            col fixed inv(i) { %N - i };
-            col fixed a = [0, 1, 0, 1, 2, 1];
-            col fixed b = [0, 0, 1, 1, 0, 5];
-            col fixed or(i) { if (a(i) != std::convert::fe(0)) || (b(i) != std::convert::fe(0)) { 1 } else { 0 } };
-            col fixed and(i) { if (a(i) != std::convert::fe(0)) && (b(i) != std::convert::fe(0)) { 1 } else { 0 } };
-            col fixed not(i) { if !(a(i) != std::convert::fe(0)) { 1 } else { 0 } };
-            col fixed less(i) { if std::convert::int(id(i)) < std::convert::int(inv(i)) { 1 } else { 0 } };
-            col fixed less_eq(i) { if std::convert::int(id(i)) <= std::convert::int(inv(i)) { 1 } else { 0 } };
+            namespace F(N);
+            let id = |i| i;
+            let inv = |i| N - i;
+            let a: int -> int = |i| [0, 1, 0, 1, 2, 1][i];
```

Comment:
just curious: what happens if the array is less than the degree here? when is it caught?

Reply:
You mean what happens if someone evaluates `a(7)`? Then we get an "array out of bounds" runtime error in the evaluator.

Reply:
Yes